pkg/cloud/ratelimit_test.go

```diff
@@ -82,3 +82,143 @@ func TestMinimumRateLimiter(t *testing.T) {
 		t.Errorf("`called` = true, want false")
 	}
 }
+
+func TestTickerRateLimiter(t *testing.T) {
+	t.Parallel()
+
+	trl := NewTickerRateLimiter(100, time.Second)
+	start := time.Now()
+	for i := 0; i < 50; i++ {
+		err := trl.Accept(context.Background(), nil)
+		if err != nil {
+			t.Errorf("TickerRateLimiter.Accept = %v, want nil", err)
+		}
+	}
+	elapsed := time.Since(start)
+	if elapsed > time.Second {
+		t.Errorf("TickerRateLimiter.Accept took too long: %v, want <1s", elapsed)
+	}
+	if elapsed < 500*time.Millisecond {
+		t.Errorf("TickerRateLimiter.Accept took too short: %v, want >500ms", elapsed)
+	}
+
+	// Use context that has been cancelled and expect a context error returned.
+	ctxCancelled, cancelled := context.WithCancel(context.Background())
+	cancelled()
+	// Verify context is cancelled by now.
+	<-ctxCancelled.Done()
+	err := trl.Accept(ctxCancelled, nil)
+	if err != ctxCancelled.Err() {
+		t.Errorf("AcceptRateLimiter.Accept() = %v, want %v", err, ctxCancelled.Err())
+	}
+}
+
+func TestCompositeRateLimiter(t *testing.T) {
+	t.Parallel()
+
+	var calledA bool
+	fa := &FakeAcceptor{accept: func() { calledA = true }}
+	arl := &AcceptRateLimiter{fa}
+	rl := NewCompositeRateLimiter(arl)
+
+	// Call default.
+	err := rl.Accept(context.Background(), nil)
+	if err != nil {
+		t.Errorf("CompositeRateLimiter.Accept = %v, want nil", err)
+	}
+	if !calledA {
+		t.Errorf("`calledA` = false, want true")
+	}
+
+	calledA = false
+	calledB := false
+	fb := &FakeAcceptor{accept: func() { calledB = true }}
+	brl := &AcceptRateLimiter{fb}
+	rl.Register("projectB", "", "", brl)
+
+	// Call registered rate limiter.
+	err = rl.Accept(context.Background(), &CallContextKey{ProjectID: "projectB"})
+	if err != nil {
+		t.Errorf("CompositeRateLimiter.Accept = %v, want nil", err)
+	}
+	if !calledB {
+		t.Errorf("`calledB` = false, want true")
+	}
+	if calledA {
+		t.Errorf("`calledA` = true, want false")
+	}
+
+	calledB = false
+	// Call default rate limiter when registered is not found
+	err = rl.Accept(context.Background(), &CallContextKey{ProjectID: "project-does-not-exist"})
+	if err != nil {
+		t.Errorf("CompositeRateLimiter.Accept = %v, want nil", err)
+	}
+	if !calledA {
+		t.Errorf("`calledA` = false, want true")
+	}
+	if calledB {
+		t.Errorf("`calledB` = true, want false")
+	}
+
+	calledA = false
+	calledC := false
+	fc := &FakeAcceptor{accept: func() { calledC = true }}
+	crl := &AcceptRateLimiter{fc}
+	rl.Register("", "networks", "", crl)
```

Comment:
you need to test that rate limiters for service and independent and does not impact on each other.

Reply:
Added missing checks